types/global.ts

```diff
@@ -240,6 +242,23 @@ export type PageStateT = {
 
 export type PageStatesT = Record<PaginateForT, PageStateT>;
 
+export type UuidT = ReturnType<typeof uuid>;
+
+export type NotificationT = {
+  message: ReactNode;
+  id: UuidT;
+  transactionHash?: string;
```

Comment:
the transaction hash is optional — the message is a react node in case you need to include other markup in there like <strong> etc. The transaction hash is used to generate another piece of markup with an etherscan link, if it exists.

we are doing different templates based on the type — success is a check mark, error is a warning icon, and pending shows a loading spinner.